bokeh/application/handlers/document_lifecycle.py

```diff
@@ -0,0 +1,67 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2017, Anaconda, Inc. All rights reserved.
+#
+# Powered by the Bokeh Development Team.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+''' Bokeh Application Handler to execute on_session_destroyed callbacks defined
+on the Document.
+
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+
+# External imports
+
+# Bokeh imports
+from .handler import Handler
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class DocumentLifeCycleHandler(Handler):
+    ''' Calls on_session_destroyed callbacks defined on the Document.
+
+    '''
+
+    safe_to_fork = True
+
+    # Public methods ----------------------------------------------------------
+
+    def modify_document(self, doc):
+        pass
```

Comment:
This needs to `return doc`, no?

Reply:
Hrm, guess not

Reply:
I don't think so, ``Application`` does this:

```python
        for h in self._handlers:
            h.modify_document(doc)
```